### PR TITLE
#35: Added regular reporting of metrics from EXA_SYSTEM_EVENTS

### LINKED ISSUE #35: Report metrics from EXA_SYSTEM_EVENTS more often
The metrics from EXA_SYSTEM_EVENTS are only written from time to time. For example one per day. For using them in Cloudwatch alarms it's however required that they are reported at least every 5 minutes.

## Solution:
report most recent value every 5 mins.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #35

### CODE REVIEW

**Comment by jakobbraun** [comment]
left open: https://github.com/exasol/cloudwatch-adapter/issues/37

File: src/main/java/com/exasol/cloudwatch/AbstractExasolStatisticsTableMetricReader.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        } catch (final SQLException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("F-CWA-13")
+                    .message("Failed to change the Exasol session time zone to UTC.").ticketMitigation().toString(),
+                    exception);
+        }
+    }
+
+    /**
+     * Get the injection free schema name. Typically this is {@code EXA_STATISTICS}. Only if in the constructor
+     * {@link #schemaOverride} was set this schema is returned.
```
Comment (morazow) [inline src/main/java/com/exasol/cloudwatch/AbstractExasolStatisticsTableMetricReader.java:95]:
```suggestion
     * Get the injection free schema name. Typically this is {@code EXA_STATISTICS}, it is different only when {@link #schemaOverride} is set in the constructor.
```

Comment (morazow) [reply src/main/java/com/exasol/cloudwatch/AbstractExasolStatisticsTableMetricReader.java:95]: Please correct the formatting.

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/cloudwatch/AbstractExasolStatisticsTableMetricReader.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    /**
+     * Read the metrics from the EXA_STATISTICS table.
+     * 
+     * @param metrics         list of metrics to read.
+     * @param start           start of the timeframe
+     * @param end             end of the timeframe
+     * @param statisticsTable EAX_STATISTICS table
+     * @return read metrics
+     */
+    protected abstract List<ExasolMetricDatum> loadMetricsForTable(final List<ExasolStatisticsTableMetric> metrics,
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/cloudwatch/AbstractExasolStatisticsTableMetricReader.java:58]: it's the interface to the different implementations of this class.

File: src/main/java/com/exasol/cloudwatch/MainExasolMetricReader.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * This class combines the different {@link ExasolMetricReader}s into one. It groups the metrics by the corresponding
+ * reader and reads them in one chunk using the corresponding reader for the group.
+ */
+public class MainExasolMetricReader implements ExasolMetricReader {
+
+    private final SimpleExasolStatisticsTableMetricReader simpleExasolStatisticsTableMetricReader;
+    private final LastValueExasolStatisticsTableMetricReader lastValueExasolStatisticsTableMetricReader;
+
+    /**
+     * loadMetricsForTable Create a new instance of {@link MainExasolMetricReader}.
```
Comment (morazow) [inline src/main/java/com/exasol/cloudwatch/MainExasolMetricReader.java:18]:
```suggestion
     * Create a new instance of {@link MainExasolMetricReader}.
```

**Review by morazow — APPROVED** [review]
Minor typo, then approved.

**Review by morazow — APPROVED** [review]

**Pull request merged**